### PR TITLE
UCP/RMA: added support of custom offset for RMA operations

### PR DESCRIPTION
**Pull request by hoopoepg**
- for some transports local access memory address and
  remote virtual addresses are not same. to support
  such cases added custom offset from requested RMA
  address

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10029/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10030/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10031/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@hoopoepg - can you please elaborate when you hit the problem ?! Does it mean that VADDR translation on the host MMU and on the Nic TLB is not the same ?

**Comment by hoopoepg** [comment]
@shamisp yes, but not only: this functionality may help to implement symmetric memory allocation across multiple processes on UCX level

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10033/ for details (Mellanox internal link).

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -945,6 +945,7 @@ typedef struct uct_rkey_bundle {
     uct_rkey_t               rkey;    /**< Remote key descriptor, passed to RMA functions */
     void                     *handle; /**< Handle, used internally for releasing the key */
     void                     *type;   /**< Remote key type */
+    uintptr_t                offset;  /**< Offset of HW address from virtual address */
```
Comment (shamisp) [inline src/uct/api/uct.h]: HW address is not exposed to user space. We would have to come up with some better clarification.

**Comment by shamisp** [comment]
@hoopoepg Can you please elaborate more, since I still missing the details. It is not clear to me why the offset gets exposed. How do you extract the information from verbs ? In case of shmem, it would have to come from upper layer ? If so how it is passed down. We don't have symmetric memory allocation API either. Are there more patches to come ?

**Pull request closed without merging**